Utilities/types.h

```diff
@@ -31,6 +31,7 @@
 #define SAFE_BUFFERS
 #define NEVER_INLINE __attribute__((noinline))
 #define FORCE_INLINE __attribute__((always_inline)) inline
+#define thread_local __thread
```

Comment:
Why for every non-msvc compiler and not just for MinGW? Also, afaik, it's not the same (rules for initialization differ from `thread_local`, though maybe it doesn't matter in our case).